### PR TITLE
TextInput refactor

### PR DESCRIPTION
**Pull request by m10l**
# Purpose of PR

Migrates the TextInput component to TypeScript.

Also:

- Creates TypeScript defs for CSS (using [typings-for-css-modules-loader](https://github.com/Jimdo/typings-for-css-modules-loader)) so that we can import CSS just using import (also has component development benefits!)
- Creates leaner builds (we were shipping stories and tests)
- Ignores `dist-storybook` directory in TypeScript config
- Adds an `inputRef` prop for setting a ref on the component

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/forma-36-react-components/src/components/TextInput/__snapshots__/TextInput.test.tsx.snap [hunk 1]
```diff
@@ -2,11 +2,10 @@
 
 exports[`renders the component as required 1`] = `
 <div
-  className="TextInput TextInput--full"
+  className=""
```
Comment (suevalov) [inline packages/forma-36-react-components/src/components/TextInput/__snapshots__/TextInput.test.tsx.snap]: Looks like this is unexpected update and something is broken with styles import

File: packages/forma-36-react-components/src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -1,36 +1,30 @@
...
+  testId?: string;
+  maxLength?: number;
+  onBlur?: (...args: any[]) => any;
+  onCopy?: (...args: any[]) => any;
+  value?: string;
+  error?: boolean;
+  required?: boolean;
+}
 
+class TextInput extends React.Component<TextInputPropTypes> {
```
Comment (suevalov) [inline packages/forma-36-react-components/src/components/TextInput/TextInput.tsx]: ```export class TextInput extends Component<TextInputPropTypes> {}```

it's the only way to build properties table properly in storybook

File: packages/forma-36-react-components/package.json [hunk 3]
```diff
@@ -120,7 +121,7 @@
     "start": "start-storybook -p 9001 -c ./tools/.storybook",
     "test": "jest",
     "build:clean": "rm -rf dist",
-    "build:scripts": "BABEL_ENV=build yarn babel src --out-dir dist --ignore *.stories.js,*.test.js --extensions '.ts','.tsx','.js'",
+    "build:scripts": "BABEL_ENV=build yarn babel src --out-dir dist --ignore 'src/**/*.stories.js','src/**/*.stories.tsx','src/**/*.test.tsx','src/**/*.test.js','src/**/*.css.d.ts' --extensions '.ts','.tsx','.js'",
```
Comment (suevalov) [inline packages/forma-36-react-components/package.json]: 👌

File: packages/forma-36-react-components/src/components/Asset/Asset.css.d.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// This file is automatically generated by typings-for-css-modules.
+// Please do not change this file!
+export interface IAssetCss {
```
Comment (suevalov) [inline packages/forma-36-react-components/src/components/Asset/Asset.css.d.ts]: Cool, I really like it

File: packages/forma-36-react-components/src/components/TextInput/__snapshots__/TextInput.test.tsx.snap (hunk 1 above)
Comment (m10l) [reply packages/forma-36-react-components/src/components/TextInput/__snapshots__/TextInput.test.tsx.snap]: Good catch, didn't need to import styles as I initially thought, fixed

File: packages/forma-36-react-components/src/components/TextInput/TextInput.tsx [hunk 5]
```diff
@@ -20,6 +20,7 @@ export interface TextInputPropTypes {
   onBlur?: (...args: any[]) => any;
   onCopy?: (...args: any[]) => any;
   value?: string;
+  inputRef?: any;
```
Comment (m10l) [inline packages/forma-36-react-components/src/components/TextInput/TextInput.tsx]: @suevalov I'm not sure what this type should be. Setting it to `any` isn't ideal, but also makes TypeScript upset in the Storybook example:

```
Property 'focus' does not exist on type '{}'.
```

Also, maybe this isn't the right way to handle refs, just seemed the simplest for most use cases.

Comment (suevalov) [reply packages/forma-36-react-components/src/components/TextInput/TextInput.tsx]: I believe it should be `RefObject<HTMLInputElement>`

Comment (suevalov) [reply packages/forma-36-react-components/src/components/TextInput/TextInput.tsx]: `import React, { RefObject } from 'react';`

File: packages/forma-36-react-components/src/components/TextInput/TextInput.tsx [hunk 6]
```diff
@@ -42,6 +43,7 @@ class TextInput extends Component<TextInputPropTypes> {
     disabled: false,
     required: false,
     width: 'full',
+    inputRef: undefined,
```
Comment (suevalov) [inline packages/forma-36-react-components/src/components/TextInput/TextInput.tsx]: listing undefined properties is optional

File: packages/forma-36-react-components/src/components/TextInput/TextInput.tsx (hunk 5 above)
Comment (m10l) [reply packages/forma-36-react-components/src/components/TextInput/TextInput.tsx]: Ahh, I also needed to update how I created the ref in the story.

```
const textInputRef = React.createRef<HTMLInputElement>();
```

**Pull request merged**